addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -116,8 +117,29 @@ def _search_valid(self, operator, value):
             or that don't need an allocation
             return [('id', domain_operator, [x['id'] for x in res])]
         """
-        date_to = self._context.get('default_date_from') or fields.Date.today().strftime('%Y-1-1')
-        date_from = self._context.get('default_date_to') or fields.Date.today().strftime('%Y-12-31')
+
+        if {'default_date_from', 'default_date_to', 'tz'} <= set(self._context):
+
+            default_date_from_str = self._context.get('default_date_to')  # sessions's local 07:00 given in UTC
+            default_date_to_str = self._context.get('default_date_from')  # sessions's local 19:00 given in UTC
+            context_timezone = pytz.timezone(self._context.get('tz'))     # user's tz
+
+            # Convert:  "2024-07-23 05:00:00"  ->  datetime.datetime(2024, 7, 23, 5, 0, tzinfo=<UTC>)
+            default_date_from_dt = datetime.datetime.fromisoformat(default_date_from_str).replace(tzinfo=pytz.UTC)
+            default_date_to_dt = datetime.datetime.fromisoformat(default_date_to_str).replace(tzinfo=pytz.UTC)
+
+            # Shift back to user's tz
+            default_date_from_dt_user_tz = default_date_from_dt.astimezone(context_timezone)
+            default_date_to_dt_user_tz = default_date_to_dt.astimezone(context_timezone)
+
+            # Cast: Datetime -> Date
+            date_from = default_date_from_dt_user_tz.strftime('%Y-%m-%d')
+            date_to = default_date_to_dt_user_tz.strftime('%Y-%m-%d')
+
+        else:
+            date_from = fields.Date.today().strftime('%Y-1-1')
```

Comment:
`today()` return the UTC current day I think, if we wanted also to get local timezone day we would need `context_today(self)` but maybe it's ok like this since it's a default

Reply:
this case is used when user's opens time off form view (img below), then we grab whole year as "requested" range and we use dates (instead of datetimes) so it should not matter

![image](https://github.com/user-attachments/assets/95497460-8e85-421e-96f8-29d6eacb3ec7)

Reply:
double checked it, yeah so the `strftime('%Y-1-1')` is literally forcing the start (end) of the year, so even if we're in a different timezone, it should not matter, I checked egde cases like start and end of the year, can't make it crash ;)

Reply:
yes, anyway it would only show the wrong year (meaning the year in utc) at 1 january so doesn't seem to matter much :+1: